lib/rules/object-shorthand.js

```diff
@@ -285,24 +285,18 @@ module.exports = {
             const fnBody = sourceCode.text.slice(arrowToken.range[1], node.value.range[1]);
 
             let shouldAddParensAroundParameters = false;
-            let tokenBeforeParams;
-
-            if (node.value.params.length === 0) {
-                tokenBeforeParams = sourceCode.getFirstToken(node.value, astUtils.isOpeningParenToken);
-            } else {
-                tokenBeforeParams = sourceCode.getTokenBefore(node.value.params[0]);
-            }
+            const tokenBeforeParams = sourceCode.getFirstToken(node.value, astUtils.isOpeningParenToken);
 
             if (node.value.params.length === 1) {
-                const hasParen = astUtils.isOpeningParenToken(tokenBeforeParams);
-                const isTokenOutsideNode = tokenBeforeParams.range[0] < node.range[0];
+                const hasParen = astUtils.isOpeningParenToken(sourceCode.getTokenBefore(node.value.params[0]));
+                const isTokenOutsideNode = tokenBeforeParams && tokenBeforeParams.range[0] < node.range[0];
 
                 shouldAddParensAroundParameters = !hasParen || isTokenOutsideNode;
             }
 
             const sliceStart = shouldAddParensAroundParameters
                 ? node.value.params[0].range[0]
-                : tokenBeforeParams.range[0];
+                : (node.value.typeParameters || tokenBeforeParams).range[0];
```

Comment:
In core rules, we only use properties defined in [the ESTree Spec](https://github.com/estree/estree/tree/master), so we shouldn't use `typeParameters`.

Reply:
Okay, I will try your suggestions.